File: Components/App.Logic/Models/SearchModel.cs
using System;
using System.Collections.Generic;

namespace App.Logic.Models
{
    public class SearchModel
    {
        public Guid Identifier { get; set; }

        public string FirstName { get; set; }

        public string LastName { get; set; }

        public string MiddleName { get; set; }

        public List<ResultModel> Results { get; set; }
    }
}

File: Web/app.client/App.App/src/app/models/SearchModel.ts
export interface SearchModel {
  identifier: string;
  firstName: string;
  lastName: string;
  middleName: string;
  results: ResultModel[];
}

type ResultModel = {
  identifier: string,
  articleIdentifier: string,
  searchIdentifier: string,
  searchText: string,
  occurrences: number,
}
